pkg/scanners/scannerv4/convert.go

```diff
@@ -117,21 +120,21 @@ func layerIndex(layerSHAToIndex map[string]int32, env *v4.Environment) *storage.
 }
 
 func vulnerabilities(vulnerabilities map[string]*v4.VulnerabilityReport_Vulnerability, ids []string) []*storage.EmbeddedVulnerability {
-	if len(vulnerabilities) == 0 {
+	if len(vulnerabilities) == 0 || len(ids) == 0 {
 		return nil
 	}
 
 	vulns := make([]*storage.EmbeddedVulnerability, 0, len(ids))
 	uniqueVulns := set.NewStringSet()
 	for _, id := range ids {
-		ccVuln, ok := vulnerabilities[id]
-		if !ok {
-			log.Debugf("Bad Input: Vuln %q from PackageVulnerabilities not found in Vulnerabilities, skipping", id)
+		if !uniqueVulns.Add(id) {
+			// Already saw this vulnerability, so ignore it.
 			continue
 		}
 
-		if !uniqueVulns.Add(ccVuln.Name) {
-			// Already added this vulnerability, so ignore it.
+		ccVuln, ok := vulnerabilities[id]
+		if !ok {
+			log.Debugf("vuln ID %q from PackageVulnerabilities not found in Vulnerabilities, skipping", id)
```

Comment:
This is possible, as I've seen when testing. It's not necessarily unexpected, as we may have filtered out some incomplete vulns. Discussing offline